msal/src/main/java/com/microsoft/identity/client/internal/controllers/MSALControllerFactory.java

```diff
@@ -126,7 +136,9 @@ public static boolean brokerEligible(@NonNull final Context applicationContext,
         }
 
         // Use broker if installed and verified
-        if (brokerInstalled(applicationContext)) {
+        if (brokerInstalled(applicationContext)
+                && (BrokerMsalController.isMicrosoftAuthServiceSupported(applicationContext)
+                || BrokerMsalController.isAccountManagerPermissionsGranted(applicationContext))) {
```

Comment:
If broker is installed w/o bound service supported, when the account manager permission is missing, should we throw the exception for missing permissions or just fallback to the msal local controller?

If not throwing the exception, the app will not know how to get unblocked and start broker calls.

If throwing the exception, what is the expected behavior when the user does not need to talk to broker to finish the auth?

Reply:
We won't know if the user needs to talk to broker to complete auth.  I'm inclined to require the application to get the permissions required in order to unblock access to the broker.  Meaning that we should throw an exception.  The design document suggested: MsalClientException.BROKER_BIND_FAILURE